RafaelRodrigues44/MerceariaAPI
# Feature/users

Pull Request: User Creation Implementation
Description

This pull request adds user creation functionality to the application. The user model, repository, controller, and Program.cs configuration were implemented to support create, read, update, and delete (CRUD) operations for users.
Main Changes

    User Model:
        Created the ApplicationUser class inheriting from IdentityUser.
        Defined the basic properties necessary for a user.

    User Repository:
        Created the IUserRepository interface with methods for user management (GetUsers, GetUserById, CreateUser, UpdateUser, DeleteUser).
        Implemented the UserRepository class using ASP.NET Identity's UserManager<ApplicationUser> to perform the operations defined in the interface.

    User Controller:
        Created the ApplicationUserController with endpoints to create, retrieve, update, and delete users.
        Utilized UserRepository for the data manipulation logic.

    Program.cs Configuration:
        Registered UserRepository and RoleRepository as services.
        Configured ASP.NET Identity to use AppDbContext.
        Configured Swagger to facilitate testing of API endpoints.

Testing

    User creation was tested through Swagger, confirming that users are correctly added to the SQLite database.
    Additional tests included retrieving all users, getting a user by ID, updating, and deleting users.

How to Test

    Run the application using dotnet run.
    Access Swagger at https://localhost:7017/swagger or http://localhost:5247/swagger.
    Use the available endpoints to create, retrieve, update, and delete users.
    Verify the responses and expected behavior of the API.

Additional Notes

    The application is configured to use an SQLite database. Ensure the connection string is correct in the appsettings.json file.
    Securing endpoints with authorization and authentication will be addressed in a future pull request.

Example User Creation Request

json

{
  "userName": "newuser",
  "email": "newuser@example.com",
  "password": "Password123!"
}
